res/controllers/midi-components-0.0.js

```diff
@@ -291,8 +291,7 @@
 
     var HotcueButton = function(options) {
         if (options.number === undefined) {
-            print("ERROR: No hotcue number specified for new HotcueButton.");
-            return;
+            throw Error("No hotcue number specified for new HotcueButton.");
```

Comment:
Who catches the exception? If Mixxx is quit, this can be a party stopper, which is a kind of no go.

Reply:
If this exception is thrown and the controller mapping does now wrap it in a try-catch-block (which would be weird), then mixxx will open a popup that a script error occurred (as we already do in other cases when the mapping throws an error, e. g. when an undefined value is called as a function).

Reply:
OK, so the party is going on, right?

Reply:
IIRC yes. Only the current "codepath" is affected. So if the error is thrown in a handler, the handler will not work properly (as expected), if the error is thrown in a constructor or the init method, the error will be shown on startup. The only case where this could break if the error is thrown in the constructor of a component and the mapping is creating Objects while running. In that case the objects wouldn't be constructed and thus the handlers of these objects won't be present. But even in this case, the rest of the mapping would still be functional iirc.

Reply:
I added this code snippet to my controller mapping:

```js
    this.someButton = new components.HotcueButton({
        group: "[Channel1]",
        midi: [0x9F, 0x02],
    });
```
Directly after loading the controller, this message popups up:
![2020-03-23-105112_806x621_scrot](https://user-images.githubusercontent.com/1834516/77304139-4f829780-6cf4-11ea-834e-161a06b8c593.png)

This is much more discoverable than using `print` which only prints to the console when `--controllerDebug` is specified (and my controller output Midi clock signals so I can only see it for the fraction of a seconds before it scrolls off-screen).